### DISCUSSION TITLE
Do we have plan to  support raft ?

### DISCUSSION THREAD

**Discussion by jcy1001** [apache/kvrocks#1619] [General]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/kvrocks/issues) and found no similar issues.

### Motivation

do we have plan to  support raft ?  when read data from slave get the same data

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

**Comment by tisonkun** [comment]
They are different algorithms and system designs. I don't think we will do that.

What do you mean by "support raft"? Is it a buzzword to you; or do you have some concrete ideas?

**Reply by tisonkun** [reply]
Briefly, it's possible like build a Redis alternative on TiKV which is based on Raft algorithm - https://medium.com/@shafreeck/titan-a-distributed-redis-protocol-compatible-nosql-database-4379a1307661 (be aware that its performance is much lower than Redis or Kvrocks, which is the main advantages of these systems)

But it's a fundamental part of system design. Unless we alter the whole system (thus, actually invent a brand new one), it's unreasonable to support it.

**Reply by mapleFU** [reply]
> But it's a fundamental part of system design. Unless we alter the whole system (thus, actually invent a brand new one), it's unreasonable to support it.

Agree

**Comment by PragmaTwice** [comment]
RedisRaft support is currently [on our roadmap](https://github.com/orgs/apache/projects/199/views/1), but TBH it is far from design or implementation.

**Reply by tisonkun** [reply]
Interesting, Redis does have a https://github.com/RedisLabs/redisraft

**Comment by mapleFU** [comment]
Maybe we can port some thing from kudu raft or other Raft implemention. But I wonder if there is anyone who really wants to use redis/kvrocks with Raft? I guess it may take time to stable, and also making writing/reading kvrocks having spike.
